### DISCUSSION TITLE
There is no guard agains different set of features for different segments

### DISCUSSION THREAD

**Discussion by d-a-bunin** [etna-team/etna#419] [Improvements]
### Description

Currently, it is possible to create a `TSDataset` with different set of features for different segments. The problem is that `TSDataset` wasn't designed for such cases and wasn't tested. I think most of features could break with such data, but removing this possibility could be a breaking change for some users.

### Thoughts

In theory, this could be declared as a bug and fixed without updating major version.

We should probably add a validation on such case during `__init__`.

We should probably add validation on such conditions into public methods which allow to change `TSDataset`:
- `add_target_components`
- `add_prediction_intervals`
- `add_columns_from_pandas`

### Related

- #310

**Comment by egoriyaa** [comment]
Solved in #581
